upgrader.go

```diff
@@ -45,12 +45,13 @@ type Upgrader struct {
 }
 
 // UpgradeListener upgrades the passed multiaddr-net listener into a full libp2p-transport listener.
-func (u *Upgrader) UpgradeListener(t transport.Transport, list manet.Listener) transport.Listener {
+func (u *Upgrader) UpgradeListener(t transport.Transport, list manet.Listener, rcmgr network.ResourceManager) transport.Listener {
```

Comment:
shouldnt it have its own rcmgr?

Reply:
Not sure. Would that be the better design?

Reply:
hrm, maybe. Shouldnt everything have the rcmgr?

Reply:
Done.